# HG changeset patch
# User Torsten <ttl@justmail.de>
# Date 1402944079 -7200
#      Mon Jun 16 20:41:19 2014 +0200
# Branch gui-release
# Node ID b43157d085baebc22cdad2938e775fefb419010e
# Parent  08d7dbd728bc15f5db4afaa08e90e2d413de66fd
make sure a new and still unnamed file is saved when run from the editor

* file-editor-tab.cc (run_file): save the file before running it when it
  is modified or when it does not have a valid file name

diff --git a/libgui/src/m-editor/file-editor-tab.cc b/libgui/src/m-editor/file-editor-tab.cc
--- a/libgui/src/m-editor/file-editor-tab.cc
+++ b/libgui/src/m-editor/file-editor-tab.cc
@@ -568,17 +568,17 @@ file_editor_tab::print_file (const QWidg
 }
 
 void
 file_editor_tab::run_file (const QWidget *ID)
 {
   if (ID != this)
     return;
 
-  if (_edit_area->isModified ())
+  if (_edit_area->isModified () | ! valid_file_name ())
     {
       _modal_dialog = true;    // force modal dialog if the file is a new one
       save_file (_file_name);  // save file dialog
       _modal_dialog = false;   // back to non-modal dialogs
     }
 
   QFileInfo info (_file_name);
   emit run_file_signal (info);
